File: frontend/src/features/inventory/services/stocktakeService.js
import axiosClient from '../../../api/axiosClient';

const stocktakeService = {
  getStocktakes: (params) => axiosClient.get('/stocktakes', { params }),
  getStatistics: () => axiosClient.get('/stocktakes/statistics'),
  getById: (id) => axiosClient.get(`/stocktakes/${id}`),
  create: (payload, complete = false) => axiosClient.post('/stocktakes', payload, { params: { complete } }),
  update: (id, payload) => axiosClient.put(`/stocktakes/${id}`, payload),
  changeStatus: (id, status) => axiosClient.put(`/stocktakes/${id}/status`, { status }),
  syncPendingMarketplaceInventory: () => axiosClient.post('/stocktakes/sync-marketplace-inventory'),
  syncStocktakeMarketplaceInventory: (id) => axiosClient.post(`/stocktakes/${id}/sync-marketplace-inventory`),
};

export default stocktakeService;

File: backend/src/main/java/fu/osms/inventory/service/StocktakeService.java
package fu.osms.inventory.service;

import fu.osms.inventory.dto.request.StocktakeSessionRequest;
import fu.osms.inventory.dto.response.StocktakeSessionResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.Map;
import java.util.UUID;

public interface StocktakeService {

    StocktakeSessionResponse createStocktake(StocktakeSessionRequest request, boolean complete);

    StocktakeSessionResponse getStocktakeById(UUID id);

    Page<StocktakeSessionResponse> getStocktakes(UUID warehouseId, String status, String keyword, Pageable pageable);

    StocktakeSessionResponse updateStocktake(UUID id, StocktakeSessionRequest request);

    StocktakeSessionResponse changeStatus(UUID id, String status);

    Map<String, Object> getStatistics();

    int syncPendingMarketplaceInventory();

    int syncStocktakeMarketplaceInventory(UUID id);
}
